addons/sale/models/sale_order.py

```diff
@@ -943,6 +943,12 @@ def action_confirm(self):
 
         return True
 
+    def _should_be_locked(self):
+        self.ensure_one()
+        # Public user can confirm SO, so we check the group on any record creator.
+        user = self[:1].create_uid
+        return user and user.sudo().has_group('sale.group_auto_done_setting')
+
```

Comment:
Only noticing now, but it seems you lost a part of the diff here, the method is not used anywhere :eyes: